From def3f0343f3cbf0431004f0b00ec0862c09ce476 Mon Sep 17 00:00:00 2001
From: "Richard W.M. Jones" <rjones@redhat.com>
Date: Thu, 25 Sep 2025 10:35:31 +0100
Subject: [PATCH] lib/types.ml: Fix formatting of debug message

With multiple disks you would see:

  virtio-blk slot 0:
        0 [scsi]virtio-blk slot 1:
        1 [scsi]virtio-blk slot 2:
        2 [scsi]virtio-blk slot 3:
        3 [scsi]

Fix this by adding a \n character.

Fixes: commit 2c96e93f63d86c9980594816a532b787532cdc2c
(cherry picked from commit dfd78446296c721efe596f2107163c5157d6c3d2)
---
 lib/types.ml | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/types.ml b/lib/types.ml
index bd4ab31c..d6709a2c 100644
--- a/lib/types.ml
+++ b/lib/types.ml
@@ -471,7 +471,7 @@ let string_of_target_bus_slots bus_name slots =
         sprintf "%s slot %d:\n" bus_name slot_nr ^
           (match slot with
            | BusSlotEmpty -> "\t(slot empty)\n"
-           | BusSlotDisk d -> string_of_source_disk d
+           | BusSlotDisk d -> string_of_source_disk d ^ "\n"
            | BusSlotRemovable r -> string_of_source_removable r ^ "\n"
           )
     ) slots in
